chia/util/paginator.py

```diff
@@ -0,0 +1,46 @@
+import dataclasses
+from math import ceil
+from typing import Sequence
+
+
+class InvalidPageSizeLimit(Exception):
+    def __init__(self, page_size_limit: int) -> None:
+        super().__init__(f"Page size limit must be one or more, not: {page_size_limit}")
+
+
+class InvalidPageSizeError(Exception):
+    def __init__(self, page_size: int, page_size_limit: int) -> None:
+        super().__init__(f"Invalid page size {page_size}. Must be between: 1 and {page_size_limit}")
+
+
+class PageOutOfBoundsError(Exception):
+    def __init__(self, page_size: int, max_page_size: int) -> None:
+        super().__init__(f"Page {page_size} out of bounds. Available pages: 0-{max_page_size}")
+
+
+@dataclasses.dataclass
+class Paginator:
+    _source: Sequence[object]
+    _page_size: int
+    _page_size_limit: int = 100
+
+    def __post_init__(self) -> None:
+        if self._page_size_limit < 1:
+            raise InvalidPageSizeLimit(self._page_size_limit)
+        if self._page_size > self._page_size_limit:
+            raise InvalidPageSizeError(self._page_size, self._page_size_limit)
+
+    def page_size(self) -> int:
+        return self._page_size
+
+    def page_size_limit(self) -> int:
+        return self._page_size_limit
+
+    def page_count(self) -> int:
+        return max(1, ceil(len(self._source) / self._page_size))
+
+    def get_page(self, page: int) -> Sequence[object]:
+        if page < 0 or page > self.page_count() - 1:
```

Comment:
```suggestion
        if 0 <= page < self.page_count():
```

Reply:
You wanna raise for good values?

Reply:
uh...  right...  i like 'em in order.  (remember when you had an approval and gave me more chances to complain?  `:]`)  but really, whatever here.

```suggestion
        if page < 0 or self.page_count() <= page:
```

Reply:
I almost like it, but what about [04d9b60](https://github.com/Chia-Network/chia-blockchain/pull/11247/commits/04d9b605c46d1f3dbd328f699a47728a0cbe249d) 😄 (swap the order)